test/distributed/fsdp/test_fsdp_meta.py

```diff
@@ -43,11 +43,15 @@
     sys.exit(0)
 
 
-def _reset_params_if_meta(is_meta, model):
+def _reset_params_if_meta(is_meta: bool, model: nn.Module):
     # For torchdistX init, we don't need to call reset_params, as
     # deferred_init(model).materialize() is equivalent to model().
     if is_meta:
-        model.reset_parameters()
+        for module in model.modules():
+            # Assume that a module has `reset_parameters()` iff it has directly
+            # managed parameters or buffers
+            if hasattr(module, "reset_parameters"):
+                module.reset_parameters()
```

Comment:
seems that the contract is now that reset_parameter is nonrecursive, did we document this somewhere?

Reply:
Yep, see the changes to the public docstrings in fully_sharded_data_parallel.py.